packages/discord.js/src/structures/User.js

```diff
@@ -151,6 +151,30 @@ class User extends Base {
     } else {
       this.avatarDecorationData = null;
     }
+
+    /**
+     * @typedef {Object} UserPrimaryGuild
+     * @property {?Snowflake} identityGuildId - The id of the user's primary guild
+     * @property {?boolean} identityEnabled - Whether the user is displaying the primary guild's tag
+     * @property {?string} tag - The user's guild tag. Limited to 4 characters
+     * @property {?string} badge - The guild tag badge hash
+     */
+
+    if (data.primary_guild) {
+      /**
+       * The primary guild of the user
+       *
+       * @type {?UserPrimaryGuild}
+       */
+      this.primaryGuild = {
+        identityGuildId: data.primary_guild.identity_guild_id,
+        identityEnabled: data.primary_guild.identity_enabled,
+        tag: data.primary_guild.tag,
+        badge: data.primary_guild.badge,
+      };
+    } else {
```

Comment:
if a patch call doesn't include primary_guild this will unset it although it should remain unchanged. Same applies to avatar_decoration_data though, no idea if that was discussed for that before. Imho it should get the same treatment as banner or accent_color did, unless we got something that guarantees this behaves differently.

Reply:
_If_ it can happen, the if check should check key in data